bioconda_utils/utils.py

```diff
@@ -176,6 +177,51 @@ def temp_os(platform):
         sys.platform = original
 
 
+def logging_run(cmds, logger=None, lvl=logging.INFO, **kwargs):
+    """
+    Like subprocess.run() with real-time STDOUT/ERR redirection to logging.
+
+    logger: Optionally pass a logger object (default: `utils` logger)
+    lvl:    Optionally set log level (default: INFO)
+
+    One log message is generated for each line received from STDERR or STDOUT.
+    Messages are prefixed with `STDERR:` or `STDOUT:` to indicate the pipe used
+    by the run process. Byte data is decoded explicitly to avoid
+    UnicodeDecodeErrors.
+
+    Raises subprocess.CalledProcessError if the exit code is non-zero.
+
+    Returns the subprocess.CompletedProcess object.
+    """
+    if logger is None:
+        logger = logging.getLogger(__name__)
+
+    p = sp.Popen(cmds, stdout=sp.PIPE, stderr=sp.PIPE,
```

Comment:
Why can't we just use the original `sp.run` and omit the pipe arguments so that all output is immediately passed to the terminal?

Reply:
Good question. I assumed that was undesired because it wasn't done in the first place.

Reply:
Hmm. As you said above, there is good reason to have the output adjusted to the scenario. Bulk builds should have very terse output. New recipes should have reasonable output. Local builds probably want it all. Passing it all though logging allows controlling the verbosity via the log level, rather than switches distributed over the code.